File: examples/api-gateway-v2/health.js
// Simple health check handler — tests multi-function routing with v2
exports.handler = async (event) => {
  console.log("Health check — v2 event version:", event.version);

  return {
    statusCode: 200,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      status: "healthy",
      timestamp: new Date().toISOString(),
      eventVersion: event.version || "unknown",
      routeKey: event.routeKey,
    }),
  };
};

File: examples/terraform-modules/modules/api/routes/health/health.py
import json
import os

def handler(event, context):
    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({
            "status": "healthy",
            "service": os.environ.get("SERVICE_NAME", "unknown"),
            "environment": os.environ.get("ENV", "unknown"),
        }),
    }
